pkg/server/server.go

```diff
@@ -464,33 +467,40 @@ func (s *Server) IsReady(ctx context.Context) (bool, error) {
 	return s.datastore.IsReady(ctx)
 }
 
-// resolveAuthorizationModelID takes a modelId. If it is empty, it will find and return the latest authorization model ID.
+// resolveAuthorizationModelID takes a modelID. If it is empty, it will find
+// and return the latest authorization modelID. If is not empty, it will
+// validate it and return it.
 //
-// If is not empty, it will validate it and return it.
-//
-// This allows caching of types. If the user inserts a new authorization model and doesn't
-// provide this field (which should be rate limited more aggressively) the in-flight requests won't be
-// affected and newer calls will use the updated authorization model.
+// This allows caching of types. If the user inserts a new authorization model
+// and doesn't provide this field (which should be rate limited more
+// aggressively) the in-flight requests won't be affected and newer calls will
+// use the updated authorization model.
 func (s *Server) resolveAuthorizationModelID(ctx context.Context, store, modelID string) (string, error) {
 	ctx, span := tracer.Start(ctx, "resolveAuthorizationModelID")
 	defer span.End()
 
+	defer func() {
+		span.SetAttributes(attribute.KeyValue{Key: authorizationModelIDKey, Value: attribute.StringValue(modelID)})
+		grpc_ctxtags.Extract(ctx).Set(authorizationModelIDKey, modelID)
+		_ = grpc.SetHeader(ctx, metadata.Pairs(AuthorizationModelIDHeader, modelID))
+	}()
+
```

Comment:
this is changing behavior - now, the auth model ID header is being set even if this function errors. Is there a test for this?

also, can we not introduce this new `defer func` pattern until we have agreed with the team that we want it? at least for me, i'm not used to it, so it takes me time to reason through it